python/ray/experimental/sgd/pytorch/pytorch_trainer.py

```diff
@@ -74,67 +76,82 @@ def __init__(self,
                  "https://github.com/pytorch/examples/issues/467."))
 
         self.model_creator = model_creator
+        self.data_creator = data_creator
         self.train_function = train_function
+        self.optimizer_creator = optimizer_creator
+        self.loss_creator = loss_creator
         self.validation_function = validation_function
+        self.initialization_hook = initialization_hook
         self.config = {} if config is None else config
         self.optimizer_timer = utils.TimerStat(window_size=1)
 
         if backend == "auto":
             backend = "nccl" if use_gpu else "gloo"
 
         logger.info("Using {} as backend.".format(backend))
-
+        self.backend = backend
+        self.use_gpu = use_gpu
+        self.batch_size = batch_size
+        self.max_replicas = num_replicas
+        self.tmpdir = tempfile.mkdtemp(prefix="raysgd")
```

Comment:
`self.temp_directory` or at least `temp_dir` for consistency.

Reply:
done